app/assets/javascripts/discourse/app/models/category-list.js

```diff
@@ -103,7 +104,7 @@ CategoryList.reopenClass({
     return ajax(
       `/categories.json?parent_category_id=${category.get("id")}`
     ).then((result) => {
-      return CategoryList.create({
+      return EmberObject.create({
```

Comment:
This was a weird one. `CategoryList` extends from ArrayProxy, so it really doesn't make sense to add random properties to it. The biggest problem is that, if the 'real' array (i.e. the `content` property of an ArrayProxy) is empty, then the entire ArrayProxy instance is considered 'falsey' in handlebars templates.

So, without this change, doing

```
{{#if categoryListInstance}}
  exists
{{/if}}
```
does not work 🤯 

Elsewhere, we do use `CategoryList` properly as an array, so that's why the change is isolated to this `listForParent` function.